### DISCUSSION TITLE
Next JS static js chunk undefined build issue

### DISCUSSION THREAD

**Discussion by akshaymhk** [vercel/next.js#38737] [Help]
Sometimes when we build project using npm build, project builds successfully without any errors. But when opening webpages, sometimes a chunk is requested by the webpage having an undefined hash (i.e **https://static1.mywebsitecdn.com/cdnnext/_next/static/chunks/8204.undefined.js**). Due to which, page is unable to open and sometimes next/link stops working without any errors or warning.

I have checked my **build-manifest.json**, entry for the **8204** chunk is missing in this file. And searched .next folder no file available with this name **8204** .

I have multiple servers running behind LB. So, I am using custom build id in **next.config.js** (i.e **generateBuildId**) as suggested in this nextjs doc link https://nextjs.org/docs/api-reference/next.config.js/configuring-the-build-id . I am using the git commit hash as BUILD_ID. 

I am just not able to understand why this issue happening sometimes. and will be fixed if we just change the commit hash by adding a simple space in code.

And sometimes same Build_id works on a server and gave the above issue on another server.

**Environment :** 
OS - Linux
Next.js version - 11.0.0
Webpack - 5 (default used by next Next.js)

**Expected :** 
**https://static1.mywebsitecdn.com/cdnnext/_next/static/chunks/8204.somehash.js** . no undefined in chunk hash.

**Comment by ssipos90** [comment]
Hi. Have you found a fix for this?

**Reply by GabrielCoutz** [reply]
Same in Next 14.1.0. After use code-splitting.

**Reply by MedZH-SPACE** [reply]
did you find a solution for this type of problem ?

**Reply by BohdanYavorskyi** [reply]
@akshaymhk @GabrielCoutz @ssipos90 have you found the root cause for it?

**Comment by ytham** [comment]
Just ran into this issue today on Next 14.1.0. :(

**Reply by MedZH-SPACE** [reply]
did you find a solution ?

**Reply by BohdanYavorskyi** [reply]
@ytham @ZOUHAIDI-Mohamed-Ettayeb  have you found a fix/root cause?
